spring-native-configuration/src/main/java/org/springframework/data/JpaComponentProcessor.java

```diff
@@ -70,6 +72,11 @@ public void process(NativeContext imageContext, String componentType, List<Strin
 
 		Type domainType = imageContext.getTypeSystem().resolveName(componentType);
 		typeProcessor.use(imageContext).toProcessType(domainType);
+
+		for (String listener : domainType.findAnnotationValue(ENTITY_LISTENERS, false, false)) {
+			Type listenerType = imageContext.getTypeSystem().Lresolve(listener);
+			imageContext.addReflectiveAccess(listenerType.getDottedName(), AccessBits.FULL_REFLECTION);
```

Comment:
Does it make sense to be more specific here in terms of methods that require access? 
Maybe something like:
```java
List<MethodDescriptor> methodDescriptors = listenerType
	.getMethods(method -> method.hasAnnotations()) // or even just javax ones
	.stream()
	.map(it -> new MethodDescriptor(it.getName(), Arrays.asList(it.asConfigurationArray())))
	.collect(Collectors.toList());

imageContext.addReflectiveAccess(listenerType.getDottedName(), new AccessDescriptor(AccessBits.LOAD_AND_CONSTRUCT, methodDescriptors, Collections.emptyList()));
```			
Otherwise I think it's fine for now to stick with `FULL_REFLECTION`.

Reply:
> @phejl Thanks for this PR, does it take in account/solve the point raised in [#869 (comment)](https://github.com/spring-projects-experimental/spring-native/issues/869#issuecomment-879664759) ?

I asked for a separate issue.

Reply:
> Does it make sense to be more specific here in terms of methods that require access?

Thanks! Let me check that.

Reply:
@christophstrobl I've updated the PR. Thanks for your suggestion!